mypy.ini

```diff
@@ -0,0 +1,22 @@
+[mypy]
+disallow_untyped_defs = True
```

Comment:
So this will keep screaming at us until all the functions are annotated. For someone reading the report, there's a lot of noise and it's hard to find "genuine" type errors among all the "this functions is not annotated".

[`disallow_untyped_calls`](https://mypy.readthedocs.io/en/stable/config_file.html#untyped-definitions-and-calls) might be a milder, interim alternative, so we could annotate bottom-up (from the perspective of call trees) without getting outright errors.

I think I'd slightly prefer the latter, but I'm not adamant.

Reply:
I can go either way, this was just to incentivize us to get it all done.

Reply:
Ah, OK, let's go all in.